includes/admin_import.php

```diff
@@ -248,243 +371,280 @@ function tsml_import_page()
 				unset($tsml_data_sources[$_POST['tsml_remove_data_source']]);
 				update_option('tsml_data_sources', $tsml_data_sources);
 
-				tsml_alert(__('Data source removed.', '12-step-meeting-list'));
+				tsml_alert(__(' Data source removed.', '12-step-meeting-list'));
+			}
+			else {
+				tsml_alert(__(' Data source removal failed! ' . $_POST['tsml_remove_data_source'], '12-step-meeting-list'), 'error');
 			}
 		}
 
+		//set change detection test_mode setting on/off
+		if (!empty($_POST['tsml_detection_test_mode']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
+			$tsml_detection_test_mode = ($_POST['tsml_detection_test_mode'] == 'on') ? 'on' : 'off';
+			update_option('tsml_detection_test_mode', $tsml_detection_test_mode);
+			tsml_alert(__('Data Sources Debug Mode setting changed.', '12-step-meeting-list'));
+		}
+
+
 		/*debugging
 		delete_option('tsml_data_sources');
 		tsml_delete('everything');
 		tsml_delete_orphans();
 		*/
 ?>
 
```

Comment:
This is a total hack, but the only thing I've come up with that fixes the issue with the Wordpress admin footer overlapping the page.

```
<style>
	html, body {
		position: relative;
		height: auto;
	}
</style>
```